Client/Frontend/Browser/TabTrayController.swift

```diff
@@ -575,120 +349,28 @@ extension TabTrayController: PresentingModalViewControllerDelegate {
     }
 }
 
-extension TabTrayController: TabManagerDelegate {
-    func tabManager(_ tabManager: TabManager, didSelectedTabChange selected: Tab?, previous: Tab?) {
-        tabDataSource.isDragging = false
-
-        // Redraw the cells representing the selected (and recently unselected) tabs.
-        let tabs = tabDataSource.tabs
+extension TabTrayController: UIScrollViewAccessibilityDelegate {
+    func accessibilityScrollStatus(for scrollView: UIScrollView) -> String? {
+        guard var visibleCells = collectionView.visibleCells as? [TabCell] else { return nil }
+        var bounds = collectionView.bounds
+        bounds = bounds.offsetBy(dx: collectionView.contentInset.left, dy: collectionView.contentInset.top)
+        bounds.size.width -= collectionView.contentInset.left + collectionView.contentInset.right
+        bounds.size.height -= collectionView.contentInset.top + collectionView.contentInset.bottom
+        // visible cells do sometimes return also not visible cells when attempting to go past the last cell with VoiceOver right-flick gesture; so make sure we have only visible cells (yeah...)
+        visibleCells = visibleCells.filter { !$0.frame.intersection(bounds).isEmpty }
 
-        // Only redraw if there is more than one tab in the tray.
-        guard tabs.count > 1 else {
-            return
+        let cells = visibleCells.map { self.collectionView.indexPath(for: $0)! }
```

Comment:
There's no reasonable chance of this failing, is there?

Reply:
visibleCells is an array of cells on the collectionview. and then this map just looks for the indexpath of those cells. Dont see this crashing.